compiler/rustc_expand/src/mbe/macro_parser.rs

```diff
@@ -142,10 +142,13 @@ pub(super) fn compute_locs(sess: &ParseSess, matcher: &[TokenTree]) -> Vec<Match
                     locs.push(MatcherLoc::Token { token: token.clone() });
                 }
                 TokenTree::Delimited(span, delimited) => {
+                    let open_tt = TokenTree::token(token::OpenDelim(delimited.delim), span.open);
+                    let close_tt = TokenTree::token(token::CloseDelim(delimited.delim), span.close);
+
                     locs.push(MatcherLoc::Delimited);
-                    inner(sess, &[delimited.open_tt(*span)], locs, next_metavar, seq_depth);
+                    inner(sess, &[open_tt], locs, next_metavar, seq_depth);
```

Comment:
```rust
let open_token = Token::new(token::OpenDelim(delimited.delim), span.open);
```
```suggestion
                    locs.push(MatcherLoc::Token { token: open_token });
```
It should be clearer here to not use recursion and just push the token.

Reply:
True, good suggestion.